pandas/tools/merge.py

```diff
@@ -1411,6 +1411,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
             sample = objs[0]
         self.objs = objs
 
+        # Check for string axis parameter
+        if isinstance(axis, str):
+            axis = objs[0]._get_axis_number(axis)
```

Comment:
This will not work in case of concatting different Serieses with `axis='columns'`, as this axis is not defined for Series

Can you also add a test for such case?